src/main/scala/cognite/spark/v1/FlexibleDataModelRelationUtils.scala

```diff
@@ -605,14 +496,25 @@ object FlexibleDataModelRelationUtils {
         }
     }
 
+  private def validateSourceSchema(
+      source: Option[SourceReference],
+      schema: StructType,
+      propertyDefMap: Map[String, ViewPropertyDefinition]): Either[CdfSparkException, Boolean] =
```

Comment:
note: looks like propertyDefMap is empty when source=None so maybe it should be Option too and maybe bundled into a single type together with reference. Atm it looks odd that we match on source but don't really use it

Can be left for future improvements

Reply:
hmm, that's a good idea

Reply:
I'll leave it for later as you say, but I'll keep that in mind